### PR TITLE
oidc delegate version and misc changes

### PR DESCRIPTION
**Pull request by aimurphy**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: Delegate version required for OIDC added to 3 connector topics. Misc other updates discovered when looking at other mentions of OIDC.
* Jira/GitHub Issue numbers (if any): PL-50559
* Preview links/images (Internal contributors only): 

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 1]
```diff
@@ -413,27 +413,64 @@ The AWS connector has the following settings.
...
+For instructions, go to [Use IRSA](/docs/platform/connectors/cloud-providers/add-aws-connector/#use-irsa).
+
+</TabItem>
+<TabItem value="oidc" label="Use OIDC">
+
+:::note
+
+Currently, OIDC authentication for AWS connectors is behind the feature flag `CDS_AWS_OIDC_AUTHENTICATION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+
+Additionally, this option requires Harness Delegate version 24.03.82603 or later.
```
Comment (aimurphy) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:459]: Delegate version requirement added here.

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md [hunk 2]
```diff
@@ -141,18 +145,19 @@ It is possible to create a connector with a non-existent delegate. This behavior
...
+Currently, this feature is behind the feature flag `PL_GCP_OIDC_AUTHENTICATION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+
+Additionally, the OIDC connectivity mode is not compatible with Google Cloud Functions. You can't deploy Google Cloud Functions with OIDC-enabled GCP connectors.
 
 :::
 
 Select the **Connect through Harness Platform for OIDC** option to allow Harness to communicate directly with GCP through OIDC. This option uses OIDC authentication to access public cloud resources without secrets or credentials.
 
-Select the **Connect through Harness Delegate for OIDC** option to allow Harness Delegate to communicate directly with GCP through OIDC. This option uses OIDC authentication to access public cloud resources without secrets or credentials.
+Select the **Connect through Harness Delegate for OIDC** option to allow Harness Delegate to communicate directly with GCP through OIDC. This option uses OIDC authentication to access public cloud resources without secrets or credentials. This option requires Harness Delegate version 24.03.82603 or later.
```
Comment (aimurphy) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md:156]: Delegate version requirement added here.

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/kubernetes-cluster-connector-settings-reference.md [hunk 3]
```diff
@@ -349,7 +349,7 @@ For more details, go to [Managing Service Accounts](https://kubernetes.io/docs/r
 
 #### OpenID Connect
 
-These settings come from your OIDC provider authorization server or the provider app you use to log in.
+These settings come from your OIDC provider authorization server or the provider app you use to log in. This option requires Harness Delegate version 24.03.82603 or later.
```
Comment (aimurphy) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/kubernetes-cluster-connector-settings-reference.md:352]: Delegate version requirement added here.

**Review by brian-f-harness — APPROVED** [review]

**Comment by aimurphy** [comment]
I am going to merge this since tomorrow is my last day. If you disagree with any of my changes, please feel free to make modifications in other PRs.

**Pull request merged**